docs/cloudformation_compatibility.rst

```diff
@@ -0,0 +1,97 @@
+CloudFormation Compatibility Section
+===============
+
+.. contents::
+
+SAM is built ontop of CloudFormation Transforms. Therefore, we need to support different CloudFormation Capabilities like: Attributes, Intrinsic functions, etc.
+
+CloudFormation Resources Attributes
+
+======================== ========================
+     Attribute Name             Supported?
+======================== ========================
+CreationPolicy           Not Currently
+DeletionPolicy           Not Currently
+DependsOn                `DependsOn Attribute`_
+Metadata                 Not Currently
+UpdatePolicy             Not Currently
+======================== ========================
+
+.. _DependsOn Attribute:
+
+DependsOn Attribute:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: yaml
+
+  LambdaFunction:
+    DependsOn: SomeOtherResources
+    Type: AWS::Serverless::Function
+    ...
+
+CloudFormation Intrinsic Funtions
+-------------------
+Currently, we do not support all Intrinsic Functions for all Property Values. Please see below tables for a details on which Intrinsic Functions can be used on a given field.
```

Comment:
Can you mention that intrinsics support is partial in AWS::Serverless::* resources, but fully available in other CloudFormation resources

Reply:
Will update